src/wasm-v8.cc

```diff
@@ -541,13 +496,16 @@ auto Store::make(Engine*) -> own<Store> {
 
 // Value Types
 
-struct ValTypeImpl {
+struct ValTypeImpl : public ValType {
   ValKind kind;
 
   ValTypeImpl(ValKind kind) : kind(kind) {}
-};
+  ~ValTypeImpl() { stats.free(Stats::VALTYPE, this); }
 
-template<> struct implement<ValType> { using type = ValTypeImpl; };
+  static const ValTypeImpl* from(const ValType* valtype) {
```

Comment:
Is it not possible to keep the template programming to make the conversions regular and reduce boilerplate defs like this?

Reply:
Fixed.